models/path.py

```diff
@@ -50,11 +50,12 @@ def is_valid(self, switch_a, switch_z, is_scheduled=False):
             return True
         previous = {switch_a}
         for link in self:
-            current = {link.endpoint_a.switch, link.endpoint_b.switch}
-            if len(previous & current) != 1:
+            current = {link.endpoint_a.switch, link.endpoint_b.switch} \
+                      - previous
```

Comment:
@Alopalao indeed, using the `previous` here was great. But maybe for completeness you could also ensure that it's a loop free path, keeping track of all previous visited dpids? For dynamic paths with `pathfinder` when computing the shortest path we already have that guarantee with Dijkstra's. Other than that it looks great to me, could you also explore a static path that might be looped?

Reply:
Also, just noticed a trivial typo below in the `InvalidPath` `"Previus"`, if you could also patch that up, thanks.

Reply:
Nice. Looks good. Let's close this thread.